### PR TITLE
V1/40 create presentation

### LINKED ISSUE #40: Create presentation
Component to create a new presentation

## User Story

As a User authenticated
I want to have a GUI to create my presentation
So that allow me to easily set up a presentation   

- [x] define what must be on the component
- [x] define the style, (follow new styleGuideline)
- [x] add communication with the Backend
- [x] Test the component

**Estimation: 2h**

### PR DESCRIPTION
**Pull request by SundayBraid**
Implement front end for presentation creation, this will Closes #40 ,

- Vuetify styleGuide has been implemented
- Bearer token management has been added

### CODE REVIEW

File: src/resources/js/router.js [hunk 1]
```diff
@@ -19,7 +20,14 @@ const router = new VueRouter({
       name: 'Test',
       icon: 'mdi-help-box',
       component: Hello,
-    }],
+    },
+    {
+      path: '/presentationCreation',
```
Comment (fabio10000) [inline src/resources/js/router.js]: the url should use a `-` synthax and not contain capital letters you should also follow the following rules to create url:
- `<resource>/<id?>/<action>`
in your case for the creation you should have `presentation/creer`. 
And for exemple if you need to do an edit it should look like `presentation/2/modifier`

File: src/resources/js/components/CreatePresentation.vue [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          <v-text-field
+            v-model="dataForm.date"
+            label="Choisir une date"
+            :rules="dateRules"
+            required
+            readonly
+            v-on="on"
+          />
+        </template>
+        <v-date-picker v-model="dataForm.date" @input="menu = false" />
```
Comment (fabio10000) [inline src/resources/js/components/CreatePresentation.vue]: Should be a datetime picker

File: src/resources/js/components/CreatePresentation.vue [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    titleRules: [
+      (v) => !!v || 'Vous devez renseigner un titre',
+      (v) => (v && v.length <= 20) || 'Le Nom doit contenir moins de 20 caractères',
+    ],
+    dateMenu: false,
+    dateRules: [
+      (v) => !!v || 'Vous devez renseigner une date',
+    ],
+    dataForm: {
+      date: new Date().toISOString().substr(0, 10),
```
Comment (fabio10000) [inline src/resources/js/components/CreatePresentation.vue]: the date field waits a date time see the api doc reference for more details

**Review by fabio10000 — CHANGES_REQUESTED** [review]

File: src/resources/js/components/CreatePresentation.vue [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      >
+        Créer la présentation
+      </v-btn>
+
+    </v-form>
+  </v-container>
+</template>
+
+<script>
+import axios from 'axios';
```
Comment (kayoumido) [inline src/resources/js/components/CreatePresentation.vue]: You don't need to import axios, it is already imported in `bootstrap.js`.

File: src/resources/js/components/CreatePresentation.vue [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+      if (!this.$refs.form.validate()) {
+        return;
+      }
+
+      // send http request with axios and catch response or error
+      axios.post(apiUrl, this.dataForm)
+        .then((response) => {
+          this.showMessage('success', `La presentation ${response.data.data.title} a été correctement crée`);
+          this.$refs.form.reset();
```
Comment (kayoumido) [inline src/resources/js/components/CreatePresentation.vue:105]: La pr**é**sentation <titre> a été correctement cré**ée**

**Review by kayoumido — CHANGES_REQUESTED** [review]

File: src/resources/js/router.js (hunk 1 above)
Comment (SundayBraid) [reply src/resources/js/router.js]: I changed the path name in [1a8be8](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/129/commits/1a8be8c9b35afb28b3822f0aaa9b77c82049639e) but in commented for avoiding #128

File: src/resources/js/components/CreatePresentation.vue (hunk 4 above)
Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue]: ok, it's corrected in [ea9160](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/129/commits/ea91602e50bd08ea0e882b6dc22e039e2fde1ae1)

File: src/resources/js/components/CreatePresentation.vue (hunk 5 above)
Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue:105]: sorry, My tailor is rich, corrected in [24695](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/129/commits/2469598d776166530f4306465cc53db8816f870f)

File: src/resources/js/components/CreatePresentation.vue (hunk 2 above)
Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue]: ok, there is no dateTime picker native in Vuetify so I planned to install [this](https://www.npmjs.com/package/vuetify-datetime-picker)

File: src/resources/js/components/CreatePresentation.vue (hunk 3 above)
Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue]: ok I removed ``.toISOString().substr(0, 10)`` in [00ec8](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/129/commits/00ec8640675ed1a7b02dd02ee50de5e33c2d0592)

File: src/resources/js/components/CreatePresentation.vue (hunk 2 above)
Comment (fabio10000) [reply src/resources/js/components/CreatePresentation.vue]: ok ping me back when it's done

Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue]: it was done in [b3eba8](https://github.com/kayoumido/HEIGVD-PRO-A-11/pull/129/commits/b3eba88b0637e68184221285b8fc62eae1342037)

**Review by fabio10000 — APPROVED** [review]

File: src/resources/js/components/CreatePresentation.vue [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    </v-form>
+  </v-container>
+</template>
+
+<script>
+export default {
+  data: () => ({
+    valid: true,
+    titleRules: [
+      (v) => !!v || 'Vous devez renseigner un titre',
```
Comment (kayoumido) [inline src/resources/js/components/CreatePresentation.vue:57]: Is the `!!` normal?

**Review by kayoumido — APPROVED** [review]

File: src/resources/js/components/CreatePresentation.vue (hunk 6 above)
Comment (SundayBraid) [reply src/resources/js/components/CreatePresentation.vue:57]: it's for force the conversion to boolean value

**Pull request merged**